library/ssl_cli.c

```diff
@@ -3171,7 +3171,7 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
         }
 
         /* Copy ECPoint structure to outgoing message buffer. */
-        ssl->out_msg[header_len] = own_pubkey_ecpoint_len;
+        ssl->out_msg[header_len] = (unsigned char) own_pubkey_ecpoint_len;
```

Comment:
In commit message "ssl_cli.c : add explicit casting to unsigned char", please explain what the warning was. I'm particularly curious why it wasn't warning before always using the submodule, and now is warning.

Reply:
Thanks for updating the commit message.

The commit message says "in our CI", but it's probably better to say MSVC targeting 64-bit Windows.

Any idea why we weren't seeing this problem before? When was the issue introduced?

Reply:
This code is ifdef'ed - `#if defined(MBEDTLS_USE_PSA_CRYPTO) && [...] `, so it wasn't building before on the MbedTLS repository, and since it's in ssl_cli.c, it probably wasn't build on the crypto repository either.

Reply:
OK, makes sense. I don't think we ever tested MBEDTLS_USE_PSA_CRYPTO on Windows before this PR.